cl_dll/death.cpp

```diff
@@ -262,6 +268,16 @@ int CHudDeathNotice :: MsgFunc_DeathMsg( const char *pszName, int iSize, void *p
 	DEATHNOTICE_DISPLAY_TIME = CVAR_GET_FLOAT( "hud_deathnotice_time" );
 	rgDeathNoticeList[i].flDisplayTime = gHUD.m_flTime + DEATHNOTICE_DISPLAY_TIME;
 
+	// Play kill sound
+	if (g_PlayerInfoList[killer].thisplayer &&
+		!rgDeathNoticeList[i].iNonPlayerKill &&
+		!rgDeathNoticeList[i].iSuicide &&
+		m_pCvarKillSnd->value > 0.0f)
+	{
```

Comment:
Really nice feature, I like it!
https://user-images.githubusercontent.com/5108747/157149047-2357ae6d-cdde-4c4a-b7e1-9254ed774c0a.mp4

Only thing I'd personally suggest is for the hitsound to be played when a person "I'm currently spectating" kills someone as well, like this:
```suggestion
	if ((g_PlayerInfoList[killer].thisplayer || g_iUser2 == killer) &&
		!rgDeathNoticeList[i].iNonPlayerKill &&
		!rgDeathNoticeList[i].iSuicide &&
		m_pCvarKillSnd->value > 0.0f)
	{
```
This also adds it for demo playback as can be seen in the video.